### PR TITLE
feat: enable code coverage

### PR DESCRIPTION
**Pull request by nathane**
PR adds code coverage via coveralls.io (free for OSS)

[Coverage Status](https://coveralls.io/github/laravel-shift/blueprint?branch=399/merge)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -118,4 +118,23 @@ jobs:
           command: composer require "laravel/framework:${{ matrix.laravel-version }}.*" --no-update
 
       - name: Run tests
-        run: vendor/bin/phpunit --verbose
+        run: vendor/bin/phpunit --verbose --coverage-clover clover.xml
+
+      - name: Upload coverage to Coveralls
+        env:
+          COVERALLS_PARALLEL: true
+          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (jasonmccreary) [inline .github/workflows/build.yml:126]: What is this token?

**Comment by jasonmccreary** [comment]
@nathane, I'm not a huge fan of code coverage. I think achieving 100% is ridiculous, tbh. On the flip side, however, anything less might make it feel like the project isn't "tested" (when it's actually well tested).

**Comment by nathane** [comment]
@jasonmccreary I agree with you. I was proposing we use code coverage to visually see that classes like `Tracer.php` has 0% tests. Not aiming for a number but adequately testing each class.

**Comment by jasonmccreary** [comment]
Right on. I'll merge this in the morning.

Also, I'm getting ideas for v2. Would love to get your input...

File: .github/workflows/build.yml (hunk 1 above)
Comment (nathane) [reply .github/workflows/build.yml:126]: Github's actions provide has an environment variable `GITHUB_TOKEN ` that generates an "auth-token" that lets `coveralls.io` use to "read" this public repo via Github API.  

https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token

https://docs.github.com/en/free-pro-team@latest/rest/reference/permissions-required-for-github-apps#permission-on-actions

you can also use it in actions to interact with Github's API.
```
name: Create issue on commit
on:
- push
jobs:
  create_commit:
    runs-on: ubuntu-latest
    steps:
    - name: Create issue using REST API
      run: |
        curl --request POST \
        --url https://api.github.com/repos/${{ github.repository }}/issues \
        --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' \
        --header 'content-type: application/json' \
        --data '{
          "title": "Automated issue for commit: ${{ github.sha }}",
          "body": "This issue was automatically created by the GitHub Action workflow **${{ github.workflow }}**. \n\n The commit hash was: _${{ github.sha }}_."
          }'
```

**Comment by nathane** [comment]
It might also be easier to set up [`Coveralls`](https://github.com/marketplace/coveralls) or [`Codecov`](https://github.com/marketplace/codecov) via GitHub Marketplace (and skip this PR).

**Comment by jasonmccreary** [comment]
Okay, I'll take a look over the weekend. If it is, I'll close this.

**Pull request closed without merging**